### PR TITLE
21 lowest scoring visitorhome team

### LINKED ISSUE #21: Lowest Scoring Visitor/Home Team

### PR DESCRIPTION
**Pull request by fadwil**
## Describe your changes in detail
This will update the average team score helper method and complete the lowest scoring team methods.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests
- [x] I AM MERGING WITH Westonio/futbol

## Anything else noteworthy? (E.g. note anything that might need refactoring or special attention later)
helper method will definitely need refactoring!

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -110,25 +110,50 @@ def worst_offense
...
     team_with_highest_score&.team_name
   end
 
   def highest_scoring_home_team
-    team_scores = calculate_team_scores(:home_team_id)
+    team_scores = average_team_scores("home")
     team_id_with_highest_score = team_scores.max_by { |_team_id, score| score }&.first
     team_with_highest_score = @teams.find { |team| team.team_id == team_id_with_highest_score }
     team_with_highest_score&.team_name
   end
```
Comment (westonio) [inline lib/stat_tracker.rb:124]: Thanks for updating these methods to accurately work with the helper method

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -110,25 +110,50 @@ def worst_offense
...
-
+  
   def lowest_scoring_visitor
-
+    away_scores = average_team_scores("away")
+    away_with_lowest_score = away_scores.min_by { |team_id, score| score}.first
+    team_with_lowest_score = @teams.find { |team| team.team_id == away_with_lowest_score }
+    team_with_lowest_score.team_name
```
Comment (pcbennett108) [inline lib/stat_tracker.rb:130]: Nice use of helper method and using a code block for multiple methods.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -110,25 +110,50 @@ def worst_offense
...
+        end
+    else away_or_home == "home"
+      home_data = @games.group_by { |game| game.home_team_id }
+      home_data.each do |team_id, game_info|
+        sum = game_info.sum { |game| game.home_goals }
+        game_count = game_info.count
+        home_data[team_id] = (sum / game_count.to_f).round(2)
+        end
+    end
   end
```
Comment (westonio) [inline lib/stat_tracker.rb:157]: This looks good, I agree that we'll want to try to refactor this where possible

**Review by westonio — APPROVED** [review]

**Review by pcbennett108 — APPROVED** [review]
Refactor looks good.
There might be even more methods that could use 'average_team_scores'. We will have to take a look at that once all of the methods are in.

**Pull request merged**